### PR TITLE
Deprecate Releasing to a patch kibana version workflow

### LINKED ISSUE #3406: [Refactor] Revisit `devtools` Commands, Groups and Organization
## Summary
The `devtools.py` library was originally meant to house CLI commands related to Detection Rules releases.

```
@root.group('dev')
def dev_group():
    """Commands related to the Elastic Stack rules release lifecycle."""
```

Over time, additional CLI commands have been added that are not related to the release. We should revisit these commands and groups to determine appropriate naming, functionality, grouping, etc. There is also code related to Kibana pushes for release which we no longer do that could be removed.

### LINKED ISSUE #3527: [Bug] Detection Rules Release Workflow is failing to successfully create and Integrations PR
**Describe the bug**
The [Release Workflow](https://github.com/elastic/detection-rules/actions/runs/8377758289/job/22940619096) is failing with the below error.

```
TypeError: Repository.create_pull() takes 3 positional arguments but 5 positional arguments (and 2 keyword-only arguments) were given
Error: Process completed with exit code 1.
```
The recent upgrade to python 3.12 may have cause the arugment type error.  Per the 3.12 [docs](https://pygithub.readthedocs.io/en/stable/github_objects/Repository.html#github.Repository.Repository.create_pull) the expected format would be 
```
pr = repo.create_pull(
    base=base_branch,
    head=branch_name,
    title=message,
    body=body,
    maintainer_can_modify=True,
    draft=draft
)
```

**To Reproduce**
1. Go to Actions in detection-rules repo and then to [Release fleet](https://github.com/elastic/detection-rules/actions/workflows/release-fleet.yml)
2. Click on Run workflow and supply a parameter per release steps [here](https://github.com/elastic/ia-trade-team/wiki/Detection-Rules-Releasing-v2#release-fleet-workflow)
3. Issue Build

**Expected behavior**
A clear draft PR to the integrations repository 

**Workaround**
At this step only the PR creation is failing. We can manually fetch the PR details from the build job and create them . 
Not a blocker for release steps but some manual correlations are needed like sample below

```
remote: Create a pull request for 'detection-rules/8.10.14-beta.1-98170269' on GitHub by visiting:        
remote:      https://github.com/elastic/integrations/pull/new/detection-rules/8.10.14-beta.1-98170269     
```

### PR DESCRIPTION
**Pull request by shashank-elastic**
## Issues
Resolves #3527
Resolves #3406

## Summary
- Depecrate Kibana Workflow commands in DR.
- We do not use this post 8.6 and with python upgrades we should not be maintaining this piece of code, we may encounter issues such as #3527.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by terrancedejesus** [comment]
I'd make sure to see where all of these CLI commands are referenced throughout the code, if they are, to ensure that upon deprecation we do not inadvertently effect other parts of the code.

**Comment by shashank-elastic** [comment]
> I'd make sure to see where all of these CLI commands are referenced throughout the code, if they are, to ensure that upon deprecation we do not inadvertently effect other parts of the code.

Yes definitely will update the PR with those. I wanted to unblock my other PR https://github.com/elastic/detection-rules/pull/3528 as i need that for DR release tomo, so this was a quick one to show we are tracking the deprecation and that can make it to main!

**Comment by shashank-elastic** [comment]
> I'd make sure to see where all of these CLI commands are referenced throughout the code, if they are, to ensure that upon deprecation we do not inadvertently effect other parts of the code.

We had a reference in MakeFile which was removed. I don't see any more references.

**Review by eric-forte-elastic — APPROVED** [review]
:green_circle: Manual review, looks good to me! :+1:

Great to clean up :rocket:

File: detection_rules/devtools.py [hunk 1]
```diff
@@ -356,186 +355,6 @@ def update_lock_versions(rule_ids):
     return changed
 
 
-@dev_group.command('kibana-diff')
-@click.option('--rule-id', '-r', multiple=True, help='Optionally specify rule ID')
-@click.option('--repo', default='elastic/kibana', help='Repository where branch is located')
-@click.option('--branch', '-b', default='main', help='Specify the kibana branch to diff against')
-@click.option('--threads', '-t', type=click.IntRange(1), default=50, help='Number of threads to use to download rules')
-def kibana_diff(rule_id, repo, branch, threads):
```
Comment (Mikaayenson) [inline detection_rules/devtools.py:364]: We should keep this.

Comment (eric-forte-elastic) [reply detection_rules/devtools.py:364]: Agreed!

**Review by eric-forte-elastic — CHANGES_REQUESTED** [review]
Let's add back `kibana-diff` otherwise looks good :+1:

File: detection_rules/devtools.py (hunk 1 above)
Comment (shashank-elastic) [reply detection_rules/devtools.py]: Reverted Back.

**Review by eric-forte-elastic — APPROVED** [review]
:green_circle: Manual review, looks good to me! :+1:

**Review by Mikaayenson — APPROVED** [review]
LGTM

**Review by terrancedejesus — APPROVED** [review]

**Pull request merged**